app/controllers/websocket_authentication_controller.rb

```diff
@@ -14,26 +14,20 @@ def authorize_channels
   private
 
   def authorize_job
-    match = /^job\.(?<id>\w+)$/.match(message[:channel])
-    return deny_channel unless match
-
-    job = Job.find_by(id: match[:id].to_i)
-    if job && current_user && job.user == current_user
-      accept_channel job.as_json
-    else
-      deny_channel
-    end
+    authorize_by_id("job", Job)
   end
 
   def authorize_worker_info
-    Rails.logger.info("Authorizing worker info")
-    match = /^worker_info\.(?<id>\w+)$/.match(message[:channel])
-    Rails.logger.info(match.inspect)
+    authorize_by_id("worker_info", Worker)
+  end
+
+  def authorize_by_id(channel_prefix, klass)
+    match = /^#{channel_prefix}\.(?<id>\w+)$/.match(message[:channel])
     return deny_channel unless match
 
-    worker = Worker.find_by(id: match[:id].to_i)
-    if worker && current_user && worker.user == current_user
-      accept_channel
```

Comment:
Just one comment on the diff. 

Before you had `accept_channel` for the worker, and not `accept_channel worker.as_json`. Will this new logic still have the same desired functionality for the `authorize_by_id("worker_info", Worker)` with the `instance.as_json` being passed into the `accept_channel` method?

Reply:
yeah, just sends extra info with the initial state, which we can use or ignore as we feel like on the frontend.